proposals/0001-gep.md

```diff
@@ -0,0 +1,198 @@
+---
+MEP: 1
+Title: Service Config
+Discussion: N/A
+Implementation: N/A
+---
+
+# Service Config
+
+## Abstract
+
+Glide is going to have many configs that control both functional and non-functional aspects of the service work. 
+This GEP defines how that config is going to be set, structured and maintained across the service lifetime.
+
+## Motivation
+
+Glide is a gateway that provides a bunch of configurations to set, manage, and maintain. 
+This ensures flexibility needed to meet various needs of our end users.
+
+For example:
+
+- **Telemetry Configs**: Log Output Format, Min Log Level, OTEL Configs
+- **HTTP Server Configs**: Is enabled?, IP and Port to bind, max body size, CORS, etc.
+- **SSL/TLS Configs:** Ciphers, Cert Paths, etc.
+- **LLM Provider Configs**: API Keys, Default Parameters, Timeouts, Connection Pool Configs, etc.
+- **Routing and Load Balancing Configs**: Balancing Strategy, Routing Weights, Provider Pool Definitions, etc.
+- **Caching Configs**: Redis Connection Configs, TTLs, etc.
+- **gRPC Server Configs**: Is enabled?, IP and Port to bind, etc.
+- etc.
+
+The configuration may be versioned via git to track and audit all changes. 
+At the same time, some configs are sensitive data (notably, LLM API Keys) 
+and it must not be provided directly in the config (otherwise, they would be leaked in git).
+
+As any other config, users should be able to easily apply new changes to the config. 
+The API keys may need to be rotated from time to time.
+
+Finally, the more configs we expose, the harder may become to understand the overall config
+and find the right setting to adjust. So the config structure is an important thing here to eliminate this fatigue.
+
+### Requirements
+
+- R1: Config format must be readable
+- R2: Config format must allow comments
+- R3: Config format must be `git diff`-able and versioned
+- R4: Config must be easy to understand and reason about
+- R5: Config must allow to include sensitive values from environment variables
+- R6: Config may allow to include sensitive values from other plain files
+- R7: Config should be easy to change and operate on
+
+## Design
+
+There are basically a few sources of configurations possible:
+- via Glide CLI params & args
+- via environment variables
+- via a separate config file
+- via a combination of CLI, environment variables, and/or config file
```

Comment:
How is a config file accessed by the providers?

Reply:
Config is going to be a singleton inited and loaded in the Gateway class and then passed itself or specific config like HTTP Server Config to the corresponding structs that needs it. Something like that. I did not mention that detail here as the contract is more important to decide on and discuss then to focus on how we get there under the hood.